java/client/src/main/java/glide/api/models/ClusterValue.java

```diff
@@ -0,0 +1,58 @@
+package glide.api.models;
+
+import java.util.Map;
+
+/**
+ * union-like type which can store single-value or multi-value retrieved from Redis. The
+ * multi-value, if defined, contains the routed value as a Map<String, Object> containing a cluster
+ * node address to cluster node value.
+ *
+ * @param <T> The wrapped data type
+ */
+public class ClusterValue<T> {
+    private Map<String, T> multiValue = null;
+
+    private T singleValue = null;
+
+    private ClusterValue() {}
+
+    /**
+     * Get per-node value.<br>
+     * Check with {@link #hasMultiData()} prior to accessing the data.
+     */
+    public Map<String, T> getMultiValue() {
+        assert hasMultiData();
```

Comment:
we should throw more informative errors so that the users will understand what happened.

Reply:
Updated in 1ea00600